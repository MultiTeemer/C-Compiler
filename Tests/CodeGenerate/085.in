typedef struct {
	int x, y;
} point;

typedef struct {
	point a, b;
} vector;

void print(vector a)
{
	printf("start: %d %d\n", a.a.x, a.a.y);
	printf("end: %d %d\n", a.b.x, a.b.y);
}

vector scale(vector a, int s)
{
	a.a.x *= s;
	a.a.y *= s;
	a.b.x *= s;
	a.b.y *= s;
	return a;
}

int main()
{	
	vector a, b;
	a.a.x = a.a.y = 0;
	a.b.x = a.b.y = 1;
	b.a.x = b.a.y = b.b.x = 0;
	b.b.y = 1;
	print(b);	
	print(scale(b, 3));
	print(a);
	print(scale(a, 5));
}