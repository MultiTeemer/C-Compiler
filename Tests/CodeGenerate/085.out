start: 0 0

end: 0 1

start: 0 0

end: 0 3

start: 0 0

end: 1 1

start: 0 0

end: 5 5

